File: Calories.App/Calories.App/Calories.App/Entities/Meal.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Calories.App.Entities
{
    public class Meal
    {
        public string Id { get; set; }
        public string UserId { get; set; }
        public int Calories { get; set; }
        public string Name { get; set; }
        public DateTime Time { get; set; } = DateTime.Now;
    }
}

File: server/modules/meals/entities/meal.js
const Serializer = require('calories/util/serialization-util');

class Meal {

    /** @param {Meal} init - Initial values. */
    constructor(init) {

        /**
         * @description The unique server-side id of this record.
         * @type {number}
         */
        this.id = init.id;

        /**
         * @description The id of the user that owns this record.
         * @type {string}
         */
        this.userId = init.userId;

        /**
         * @description The name of the meal (e.g. 'lunch' or 'General Tao') .
         * @type {string}
         */
        this.name = init.name;

        /**
         * @description The meal's number of calories.
         * @type {number}
         */
        this.calories = init.calories;

        /**
         * @description The meal's date and time (i.e. when it was eaten).
         * @type {Date}
         */
        this.time = Serializer.stringToDate(init.time);
    }
}

module.exports = Meal;